bugsnag-android-core/src/test/java/com/bugsnag/android/ConfigurationTest.java

```diff
@@ -180,4 +181,45 @@ public void testVersionCode() {
         configuration.setVersionCode(577);
         assertEquals(577, (int) configuration.getVersionCode());
     }
+
+    @Test
+    public void testAddMetadata() {
+        Configuration configuration = generateConfiguration();
+        configuration.addMetadata("keyVal", "foo", "bar");
+        HashMap<String, Object> testMap = new HashMap<>();
+        testMap.put("key1", "val1");
+        testMap.put("key2", "val2");
+        configuration.addMetadata("mapVal", testMap);
+
+        assertEquals("bar", configuration.getMetadata("keyVal", "foo"));
+        assertEquals("val1", configuration.getMetadata("mapVal", "key1"));
+        assertEquals("val2", configuration.getMetadata("mapVal", "key2"));
+    }
+
+    @Test
+    public void testClearMetadata() {
+        Configuration configuration = generateConfiguration();
+        configuration.addMetadata("keyVal", "foo", "bar");
+        HashMap<String, Object> testMap = new HashMap<>();
+        testMap.put("key1", "val1");
+        testMap.put("key2", "val2");
+        configuration.addMetadata("mapVal", testMap);
+
+        configuration.clearMetadata("keyVal");
+        configuration.clearMetadata("mapVal", "key2");
+
+        assertNull(configuration.getMetadata("keyVal", "foo"));
+        assertEquals("val1", configuration.getMetadata("mapVal", "key1"));
+        assertNull(configuration.getMetadata("mapVal", "key2"));
+    }
+
+    @Test
+    public void testSetUser() {
+        Configuration configuration = generateConfiguration();
+        configuration.setUser("24601", "m@rp.fr", "MM");
+        assertEquals(User.class, configuration.getUser().getClass());
```

Comment:
This assertion could be removed as it seems to test the Java type system rather than our code